### DISCUSSION TITLE
missing topics in the curriculum

### DISCUSSION THREAD

**Discussion by mrtnlhmnn** [isaqb-org/curriculum-flex#58] [Ideas]
The current topics are not addressed (sufficiently):

a) Testing for both mono/modulithic and distributed systems (another driver / requirement)
b) Migration concepts, especially when going from a mono/modulithic system to a distributed one. Concepts like blue/green or red/black deployment and replacing parts from the monolith.
c) "Chaos Engineering", as an add-on to "Resilience"
d) Concept of an inter-service/system interface ("Schnittstelle"). How to setup such an interface, interface contract etc.
e) Versioning of systems and services (like with "Semantic Versioning")

**Comment by aheusingfeld** [comment]
> a) Testing for both mono/modulithic and distributed systems (another driver / requirement)

Agreed as a MUST. I do have it in my trainings and didn't think it could be mentioned more explicitly in the curriculum

> b) Migration concepts, especially when going from a mono/modulithic system to a distributed one. Concepts like blue/green or red/black deployment and replacing parts from the monolith.

I do consider "migration" a topic for the IMPROVE curriculum which is why we already removed "migration strategies" from the curriculum in 2015. Let's rather LINK this topic explicitly to the IMPROVE curriculum

> c) "Chaos Engineering", as an add-on to "Resilience"

Agreed, let's add it as a SHOULD, @sippsack 

> d) Concept of an inter-service/system interface ("Schnittstelle"). How to setup such an interface, interface contract etc.

Can you elaborate on this one, @mrtnlhmnn? I do think this is part of "4. Integration" but I might be wrong. Thanks in advance.

> e) Versioning of systems and services (like with "Semantic Versioning")

Hm, that's a tough one. We could add it as COULD in "4. Integration", yet, I'd focus it on interface versioning and rather than on the broader services version and interoperability. LMKYT
